tests/Doctrine/Tests/OrmFunctionalTestCase.php

```diff
@@ -880,4 +869,59 @@ protected function setUpDBALTypes(): void
             Type::addType('rot13', Rot13Type::class);
         }
     }
+
+    final protected function isQueryLogAvailable(): bool
+    {
+        return $this->_em->getConnection() instanceof DbalExtensions\Connection;
+    }
+
+    final protected function enableQueryLog(): void
+    {
+        $this->getQueryLog()->enabled = true;
+    }
+
+    final protected function disableQueryLog(): void
+    {
+        $this->getQueryLog()->enabled = false;
+    }
+
+    final protected function getQueryLog(): QueryLog
+    {
+        $connection = $this->_em->getConnection();
+        if (! $connection instanceof DbalExtensions\Connection) {
+            throw new RuntimeException(sprintf(
+                'The query log is only available if %s is used as wrapper class. Got %s.',
+                DbalExtensions\Connection::class,
+                get_debug_type($connection)
+            ));
+        }
+
+        return $connection->queryLog;
+    }
+
+    /**
+     * Using the SQL Logger Stack this method retrieves the current query count executed in this test.
+     */
+    final protected function getCurrentQueryCount(): int
+    {
+        return count($this->getQueryLog()->queries);
+    }
```

Comment:
It might be worth also introducing `assertQueryCount(int $expectedCount, string $message)`

Reply:
Yeah. 🤔

Reply:
I'll do this in a follow-up.